drivers/sensor/veml7700/veml7700.c

```diff
@@ -535,30 +499,72 @@ static int veml7700_init(const struct device *dev)
 	return 0;
 }
 
+#ifdef CONFIG_PM_DEVICE
+static int veml7700_pm_action(const struct device *dev, enum pm_device_action action)
+{
+	const struct veml7700_config *conf = dev->config;
+	struct veml7700_data *data = dev->data;
+	uint8_t prev_sd;
+	int ret;
+
+	if (conf->psm != VEML7700_PSM_DISABLED) {
```

Comment:
How does software get to decide whether the device is on a power domain? It either is or it isn't.

Reply:
`psm` is a configuration value that configures the internal power saving mode of the sensor. I moved that into the `RESUME` and `SUSPEND` cases, as it's nothing to do with the `TURN_ON/OFF` event.